portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -75,29 +64,37 @@ public void jsonInit() {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // response.setContentType("text/html;");
-        // response.getWriter().println("<h1>Hello Murilo!</h1>");
-        //String[] pair = new String[2];
-        //response[0] = countries.get(4);
-        //response[1] = capitals.get(4);
-        Random rand = new Random();
+        // String json = convertArrayListToJson(hardCodedJson);
+        String json = convertToJsonUsingGson(comments);
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
What exactly happens on a POST?

When a `<form>` is submitted, by default this works like page navigation (i.e. clicking a link) and the browser will "close" the current page and show the response of the `POST`.

What should happen after a comment is posted? I would expect that the page updates/reloads with my new comment in place. This can be done by e.g. sending a "HTTP 302" redirect to tell the browser "Don't show the `/data` page, but go back to e.g. `/` (index.html)"

Your page structure has made this a little complicated, because reloading/redirecting to it won't actually show the comments until the user clicks the 'Comments' button again.

There's a few design options to deal with this:
 - All of the comments could live on a separate page (e.g. "comments.html") instead of being loaded inside of a `<div>` on the main page. (The new page would need to repeat all of the basic page structure / navigation bar / css, of course)
 - Instead of letting the browser POST the `<form>`, this can be intercepted with Javascript. Now your code is in charge of taking the form values and sending them as a POST request, and your callback can get the comment(s) in the response and update the page
 - The main page could 'remember' that the user was in the comments area (e.g. with a url hash) and load/focus it automatically. This would make the redirect work (e.g. server sends `Location: /#comments`, which triggers the javascript to load the comments right away).